### PR TITLE
Optimize `propsAPI` usage

### PR DESCRIPTION
**Pull request by gaoli**

### CODE REVIEW

File: src/components/GGEditor/index.js [hunk 1]
```diff
@@ -23,6 +23,18 @@ class GGEditor extends React.Component {
     Global.set('trackable', Boolean(value));
   }
 
+  static withPropsAPI(WrappedComponent) {
```
Comment (tennisonchan) [inline src/components/GGEditor/index.js]: why don't we separate `withPropsAPI` into its own file?

File: src/components/Adapter/propsAPI.js [hunk 2]
```diff
@@ -5,15 +5,19 @@ class PropsAPI {
...
-  findItem = id => this.activePage.find(id);
+  read = data => this.currentPage.read(data);
+  save = () => this.currentPage.save();
+
+  add = (type, model) => this.currentPage.add(type, model);
+  find = id => this.currentPage.find(id);
+  update = (item, model) => this.currentPage.update(item, model);
+  remove = item => this.currentPage.remove(item);
+
+  getSelected = () => this.currentPage.getSelected();
```
Comment (tennisonchan) [inline src/components/Adapter/propsAPI.js]: it seems we are just using exposing certain API on `currentPage`. 

Maybe we can do something like this in `constructor`:
```js
['read', 'save'].forEach((key) => {
  this[key] = this.currentPage[key].bind(this)
});
```

or this
```js
read = this.currentPage.bind(this)
```

File: src/components/GGEditor/index.js [hunk 3]
```diff
@@ -23,6 +23,18 @@ class GGEditor extends React.Component {
...
+    return class extends React.Component {
+      static contextTypes = {
+        propsAPI: PropTypes.object,
+      };
+
+      render() {
+        return <WrappedComponent propsAPI={this.context.propsAPI} {...this.props} />;
```
Comment (tennisonchan) [inline src/components/GGEditor/index.js]: We do it explicitly in this way, or we want to `<WrappedComponent {...this.context.propsAPI} {...this.props} /?`

Comment (gaoli) [reply src/components/GGEditor/index.js]: 建议挂载 `propsAPI` 属性之下，防止命名冲突

File: src/components/Adapter/propsAPI.js (hunk 2 above)
Comment (gaoli) [reply src/components/Adapter/propsAPI.js]: Good idea 👍

File: src/components/GGEditor/index.js (hunk 1 above)
Comment (gaoli) [reply src/components/GGEditor/index.js]: Thank you for you suggestion !

**Comment by gaoli** [comment]
fix #30

**Pull request merged**